src/gfn/gym/hypergrid.py

```diff
@@ -86,6 +86,7 @@ def update_masks(self, states: type[DiscreteStates]) -> None:
         """Update the masks based on the current states."""
         # Not allowed to take any action beyond the environment height, but
         # allow early termination.
+        # TODO: do we need to handle the conditional case here?
```

Comment:
I don't think so. We may need another EnvBase that incorporates conditions.

Reply:
ConditionalEnvs are interesting. I should actually do this ASAP so we have a working conditional GFN example.